suite-native/module-home/src/screens/HomeScreen.tsx

```diff
@@ -14,13 +19,22 @@ const homeScreenContentStyle = prepareNativeStyle(_ => ({
 
 export const HomeScreen = () => {
     const { applyStyle } = useNativeStyles();
+
+    const isPortfolioEmpty = useSelector(selectIsAccountsPortfolioEmpty);
+    const isDiscoveryActive = useSelector(selectIsDeviceDiscoveryActive);
+    const { isUsbDeviceConnectFeatureEnabled } = useIsUsbDeviceConnectFeatureEnabled();
+
     return (
         <Screen>
-            <VStack spacing="large" style={applyStyle(homeScreenContentStyle)}>
-                <PortfolioGraph />
-                <Assets />
-                <DashboardNavigationButtons />
-            </VStack>
+            {isPortfolioEmpty && !isDiscoveryActive ? (
```

Comment:
You have this `if` here and then lot same check for `isDiscoveryActive` also in Graph components that should be never rendered according to this so it's redundant to have this check in nested components.

Reply:
019dcf358